planner/core/task.go

```diff
@@ -1967,7 +1976,7 @@ func RemoveUnnecessaryFirstRow(
 				}
 				if gbyExpr.Equal(sctx, aggFunc.Args[0]) {
 					canOptimize = true
-					funcMap[aggFunc].Args[0] = finalGbyItems[j]
+					firstRowFuncMap[aggFunc].Args[0] = finalGbyItems[j]
```

Comment:
`gbyExpr.Equal()` is judged by the number semantic？if yes, '00' = 'gg'?

Reply:
The `Equal` function is an attribute method of the Expression interface.
It seems not judged by the number semantic.

BTW, it is not modified by this commit.